pkg/inject/inject.go

```diff
@@ -305,24 +306,33 @@ func (conf *ResourceConfig) GetPodPatch(injectProxy bool) ([]byte, error) {
 	return res, nil
 }
 
-// GetServicePatch returns the JSON patch containing the service opaque ports
-// annotation if the annotation is present on the namespace, but absent on the
-// service.
-func (conf *ResourceConfig) GetServicePatch() ([]byte, error) {
-	_, ok := conf.workload.Meta.Annotations[k8s.ProxyOpaquePortsAnnotation]
-	// There does not need to be a patch if the service already has the
-	// annotation.
+// AnnotateOpaquePorts returns two values. The first value is the the opaque
+// ports annotation value. The second is used to decide whether or not the
+// caller should add the annotation. The caller should not add the annotation
+// if the resource already has its own.
+func (conf *ResourceConfig) AnnotateOpaquePorts() (string, bool) {
```

Comment:
So we do need both return values here. The reason is that if the workload is annotated with `config.linkerd.io/opaque-ports: ""`, then the user is intentionally setting no opaque ports. In this case `annotation == ""` and `opaquePortsOk == true`.

So we need to be able to differentiate between no annotations and explicitly empty annotations.

`AnnotateOpaquePorts` could be renamed since it's just getting the annotation. I lean towards `GetOpaquePorts` since it's a little more in line with the return style of `(string, bool)`.

Reply:
@dadjeibaah Renamed 👍